babel_monitor/src/lib.rs

```diff
@@ -223,6 +227,38 @@ impl<T: Read + Write> Babel<T> {
         }
         Ok(vector)
     }
+
+    /// In this function we take a route snapshot then loop over the routes list twice
```

Comment:
as much as I complain about the speed of this at 2*routes time it's within constant time of the fastest it could be (assuming we don't make babel sort routes somehow so we can search them more efficiently). So it can probably stay unless there's an obvious way to get the neighbor ip's I'm missing.

Reply:
One thing we might want to do in the future is have all babel methods operate over some sort of BabelSnapshot, so we don't need to parse routes/hit babel every time